compiler/src/iree/compiler/Codegen/TransformStrategies/GPU/AbstractGemmLikeStrategy.h

```diff
@@ -135,26 +132,38 @@ struct AbstractGemmLikeStrategy {
 
   // wmma preconditions that we want to lift out in an actionnable top-level
   // error message instead of failing late in the transformation schedule.
-  // TODO: These are now hardcoded for f32 but are element-type dependent.
-  // Precondition: the pipeline transformation for wmma requires at least 2
-  // k-groups.
+  // TODO: These are now hardcoded for f16 and f32 but are element-type
+  // dependent. Precondition: the pipeline transformation for wmma requires at
+  // least 2 k-groups.
   constexpr static int64_t kMinWmmaMinM = 16;
   constexpr static int64_t kMinWmmaMinN = 16;
-  constexpr static int64_t kMinWmmaMinK = 8;
+  int64_t kMinWmmaMinK() const { return lhsElementalBitWidth == 32 ? 8 : 16; }
 
   // mma.sync preconditions that we want to lift out in an actionnable top-level
   // error message instead of failing late in the transformation schedule.
-  // TODO: These are now hardcoded for f32 but are element-type dependent.
-  // Precondition: the pipeline transformation for mma.sync requires at least 2
-  // k-groups.
+  // TODO: These are now hardcoded for f16 and f32 but are element-type
+  // dependent. Precondition: the pipeline transformation for mma.sync requires
+  // at least 2 k-groups.
   constexpr static int64_t kMinMmaSyncGroups = 2;
   // Precondition: the pipeline transformation for mma.sync requires at least a
   // pipeline depth of 3.
   constexpr static int64_t kMinMmaSyncPipelineDepth = 3;
-  // Precondition: if mma.sync is used, the tile sizes must be at least 8x8x4.
+  // Precondition: if mma.sync is used, the tile sizes must be at least 8x8x4
+  // for f32 and 8x8x8 for f16.
   constexpr static int64_t kMinMmaSyncMinM = 8;
   constexpr static int64_t kMinMmaSyncMinN = 8;
-  constexpr static int64_t kMinMmaSyncMinK = 4;
+  int64_t kMinMmaSyncMinK() const { return lhsElementalBitWidth == 32 ? 4 : 8; }
+
+  // cp.async requires 4, 8, or 16 total bytes, reflected here by the minimum
+  // required vector sizes for smaller bit widths.
```

Comment:
Seems like <32 element bit widths have >1 required vector sizes for asynchronous copies ([from this requirement](https://github.com/llvm/llvm-project/blob/7ac0ff562ab8b0c2134b17333f4efa6fbbeab48c/mlir/lib/Conversion/NVGPUToNVVM/NVGPUToNVVM.cpp#L395)). In conjunction with [this comment](https://github.com/openxla/iree/pull/14138#issuecomment-1594865289) I'm guessing we won't always be able to use asynchronous copies for unaligned sub-32 bit types so I tried to allow the strategy to fail a bit more gracefully on an invalid configuration for now.

Reply:
as mentioned offline, feel free to can patch the subset of #14184 that you need to accelerate yourself

Reply:
I opened #14192 for this; will rebase on top of that and drop this check.